### PR TITLE
Feature/dop 1204 implement project ids

### PR DESCRIPTION
**Pull request by Jorg88**

### CODE REVIEW

**Review by tunaman — APPROVED** [review]

File: src/mpyl/schema/project.schema.yml [hunk 1]
```diff
@@ -35,6 +36,10 @@ properties:
       type: string
     uniqueItems: true
     minItems: 1
+  projectId:
+    description: >-
+      The rancher project id that this project should be deployed to, required for rbac. Represents a domain in rancher.
+    $ref: '#/definitions/dtapValue'
```
Comment (koeves) [inline src/mpyl/schema/project.schema.yml:42]: I would move this to the kubernetes section at least

Comment (koeves) [reply src/mpyl/schema/project.schema.yml:42]: Maybe even a specific kubernetes.rancher section

File: src/mpyl/project.py [hunk 2]
```diff
@@ -308,8 +308,18 @@ def from_config(values: dict):
         )
 
 
+@dataclass(frozen=True)
+class Rancher:
+    project_id: str
```
Comment (koeves) [inline src/mpyl/project.py:313]: Okay, one last comment about this I promise :D 
Why is this mandatory? From an mpyl perspective I mean

Comment (Jorg88) [reply src/mpyl/project.py:313]: It's not really, only if you have `kubernetes.deployment` and it defaults to an empty string.

Comment (koeves) [reply src/mpyl/project.py:313]: Shouldn't we still be able to provide a default project ID per env though? I mean, we used it like that before and now we want to be able to set the projectId per project, but setting the project IDs per environment also makes sense (even though we want to do something else now)

Comment (koeves) [reply src/mpyl/project.py:313]: My question was whether we _need_ to set the project ID for every project if it uses Kubernetes Deploy. From an mpyl's perspective, I don't think the project ID should be mandatory per project

**Review by koeves — APPROVED** [review]

**Pull request merged**